src/nvim/getchar.c

```diff
@@ -3925,15 +3960,29 @@ static char_u *eval_map_expr(char_u *str, int c)
   save_cursor = curwin->w_cursor;
   save_msg_col = msg_col;
   save_msg_row = msg_row;
-  p = eval_to_string(expr, NULL, false);
+  if (mp->m_luaref > 0) {
+    Error err = ERROR_INIT;
+    Array args = ARRAY_DICT_INIT;
+    if (mp->m_expr) {
+      Object ret = nlua_call_ref(mp->m_luaref, NULL, args, true, &err);
+      if (ret.type == kObjectTypeString) {
+        p = (char_u *)xstrndup(ret.data.string.data, ret.data.string.size);
```

Comment:
Should we apply `replace_termcodes` on result of function ?

Reply:
Too magic. I'd rather add a shortcut for `vim.api.nvim_replace_termcodes(str, true, true, true)` (like `vim.termcode(str)` or something ) in the next PR.